stm32-modules/heater-shaker/simulator/system_thread.cpp

```diff
@@ -66,10 +67,31 @@ struct system_thread::TaskControlBlock {
     SimSystemTask task;
 };
 
+template <size_t N>
```

Comment:
I know it's a tiny function, but can't it be in `include/common/simulation`  rather than duplicated?

Reply:
Tbh I was avoiding setting up a `common-simulator` target... will set that up now